### DISCUSSION TITLE
documentation for lock file formats

### DISCUSSION THREAD

**Discussion by konstin** [python-poetry/poetry#6763] [General] [answered]
is there any documentation for the lock file format versions? i've just seen in the release notes that there will be a lock file format 2.0, and i was wondering what the differences are. i'm asking because for [monotrail](https://github.com/konstin/poc-monotrail) i have to read poetry.lock to get the resolved set of dependencies and their filenames/hashes. Currently, i've just [hardcoded the poetry 1.2 format](https://github.com/konstin/poc-monotrail/blob/9cff50d15f5e87387a0e70e0d312a3ffd6c16878/src/poetry_integration/poetry_lock.rs#L8-L161).

**Comment by neersighted** [comment, marked as answer]
There is no standalone documentation beyond the source code at this time. The lock file 2.0 format moves the files from a global table, to instead be per `[[package]]` entry as `package.files = [{file objects}...]`.
